### PR TITLE
Refactor tBTC flow

### PR DESCRIPTION
**Pull request by valentinoConti**
### Description

This PR implements a refactor on the Threshold tBTC flow.

Before, we were allowing multiple wrapped tBTCs token to get generated: wTBTC-Eth, wTBTC-Opt, wTBTC-Arb and wTBTC-Pol. Those tokens were not 100% interchangeable between all the chains. It was possible to end up with double wrapped tokens.

Now, the only wrapped asset that is going to circulate on portal bridge is the wTBTC-Eth. If the source chain it's a threshold-supported chain like Optimism, and from there you send canonical tBTC to, for example, Avalanche. Instead of receiving wTBTC-Opt you will receive wTBTC-Eth. And you can swap that wTBTC-Eth for canonical tBTCs on any supported chain (be Optimism or any other)

Basically, you can send your tBTC from any chain to any other chain and you should never see two assets nor have any double-wrapped asset.

### CODE REVIEW

**Review by sebastianscatularo — APPROVED** [review]

File: src/hooks/useHandleTransfer.tsx [hunk 1]
```diff
@@ -308,207 +321,132 @@ async function evm(
...
-            receipt,
-            getBridgeAddressForChain(chainId)
-          );
-          const emitterAddress = getEmitterAddressEth(
-            getTokenBridgeAddressForChain(chainId)
-          );
+      const tx = await L2WormholeGateway.sendTbtc(
+        amountNormalizeAmount,
+        recipientChain,
+        recipientAddress,
```
Comment (raop155) [inline src/hooks/useHandleTransfer.tsx:381]: Just curious, now there's not need for any type of conversion, why?

File: src/hooks/useHandleTransfer.tsx [hunk 2]
```diff
@@ -794,8 +752,9 @@ async function injective(
       asset,
       transferAmountParsed.toString(),
       targetChain,
-      targetAddress,
-      feeParsed.toString()
+      additionalPayload?.receivingContract || targetAddress,
+      feeParsed.toString(),
+      additionalPayload?.payload
```
Comment (raop155) [inline src/hooks/useHandleTransfer.tsx:757]: Question:
Why do we pass an additional `additionalPayload?.payload` param here, and in others `transferFromX` methods?

File: src/hooks/useHandleTransfer.tsx (hunk 1 above)
Comment (valentinoConti) [reply src/hooks/useHandleTransfer.tsx:381]: Before, I was using the `readableTargetAddress` which was fine for EVMs but not for other chains.

`recipientAddress` it's already processed and it's always a valid 32-bytes address 💪🏼 so using this there's no need to do any conversions

File: src/hooks/useHandleTransfer.tsx [hunk 3]
```diff
@@ -308,207 +321,132 @@ async function evm(
...
+      const additionalPayload = maybeAdditionalPayload();
+
+      receipt = isNative
         ? await transferFromEthNative(
             getTokenBridgeAddressForChain(chainId),
             signer,
             transferAmountParsed,
             recipientChain,
-            recipientAddress,
+            additionalPayload?.receivingContract || recipientAddress,
```
[the remark this answers is not in the document]
Comment (valentinoConti) [reply src/hooks/useHandleTransfer.tsx:405]: Exactly, this is to access a different address than the default one, and also to add an additional payload if needed.

For now, it will be available only for TBTC flow, but shortly SEI stuff will be added to the same method as you can see on this [draft SEI PR](https://github.com/XLabs/portal-bridge-ui/pull/157) (I'll have to sync it with main branch after the merge of this one) 😄

File: src/hooks/useHandleTransfer.tsx (hunk 2 above)
Comment (valentinoConti) [reply src/hooks/useHandleTransfer.tsx:757]: So if later we need to send an additional payload (or send a tx to a different contract) on any chain we'll just need to add the condition into the maybeAdditionalPayload function 👌🏼
